src/plugins/analysis/file_system_metadata/code/file_system_metadata.py

```diff
@@ -36,85 +58,67 @@ class StatResult(NamedTuple):
-    VERSION = '0.2.1'
-    TIMEOUT = 600
-    FILE = __file__
-
-    ARCHIVE_MIME_TYPES = [  # noqa: RUF012
-        'application/gzip',
-        'application/x-bzip2',
-        'application/x-tar',
-    ]
-    FS_MIME_TYPES = [  # noqa: RUF012
-        'filesystem/btrfs',
-        'filesystem/cramfs',
-        'filesystem/dosmbr',
-        'filesystem/ext2',
-        'filesystem/ext3',
-        'filesystem/ext4',
-        'filesystem/hfs',
-        'filesystem/jfs',
-        'filesystem/minix',
-        'filesystem/reiserfs',
-        'filesystem/romfs',
-        'filesystem/udf',
-        'filesystem/xfs',
-        'filesystem/squashfs',
-    ]
 
-    def __init__(self, *args, **kwargs):
-        self.db = DbInterfaceCommon()
-        super().__init__(*args, **kwargs)
-
-    def process_object(self, file_object: FileObject) -> FileObject:
-        result = self._extract_metadata(file_object)
-        if 'files' in result:
-            self._add_tag(file_object, result['files'])
-        result['contained_in_file_system'] = self._parent_has_file_system_metadata(file_object)
-        file_object.processed_analysis[self.NAME] = result
-        return file_object
-
-    def _parent_has_file_system_metadata(self, file_object: FileObject) -> bool:
-        if hasattr(file_object, 'temporary_data') and 'parent_fo_type' in file_object.temporary_data:
-            return self._has_correct_type(file_object.temporary_data['parent_fo_type'])
-        return self._parent_fo_has_results(file_object)
-
-    def _parent_fo_has_results(self, file_object: FileObject) -> bool:
-        for parent_uid in file_object.parents:
-            analysis_entry = self.db.get_analysis(parent_uid, 'file_type')
-            if analysis_entry is not None and self._has_correct_type(analysis_entry['result']['mime']):
-                return True
-        return False
-
-    def _has_correct_type(self, mime_type: str) -> bool:
-        return mime_type in self.ARCHIVE_MIME_TYPES + self.FS_MIME_TYPES
-
-    def _extract_metadata(self, file_object: FileObject) -> dict:
-        file_type = file_object.processed_analysis['file_type']['result']['mime']
-        if file_type in self.FS_MIME_TYPES:
-            return self._extract_metadata_from_file_system(file_object)
-        if file_type in self.ARCHIVE_MIME_TYPES:
-            return self._extract_metadata_from_tar(file_object)
-        return {}
-
-    def _extract_metadata_from_file_system(self, file_object: FileObject) -> dict:
+    class Schema(BaseModel):
+        files: List[FileMetadata]
+
+    def __init__(self):
+        metadata = self.MetaData(
+            name=self.NAME,
+            dependencies=['file_type'],
+            description=(
+                'extract file system metadata (e.g. owner, group, etc.) from file system images contained in firmware'
+            ),
+            version='1.0.0',
+            timeout=600,
```

Comment:
The plugin works on file systems/archives. This means that it may need to unpack and analyze thousands of files. From experience, this may take a couple of minutes.

Reply:
> it uses the default timeout

While I know that we disagree on the default timeout, I think we should lower the timeout for this plugin as a result of your tests.